### ISSUE TITLE
Sometimes, osqueryd quits automatically in Windows platform

### ISSUE THREAD

**Issue by christmasliu** [osquery/osquery#7200]
# Bug report

### What operating system and version are you using?
Windows 10 19043

### What version of osquery are you using?
OSQuery 4.9

### What steps did you take to reproduce the issue?
Register as a OSQuery extension plugin and launch osqueryd to do scheduled query. Run it for several hours and found osqueryd quits. This issue is not noticed before.

### What did you expect to see?
osqueryd keep running.

### What did you see instead?
osqueryd quits.

**Comment by Smjert** [comment]
@christmasliu thanks for the report!

Do you have logs from osquery that might shed some light on what's happening?
Have you tried running osquery with the `--verbose` flag to increase the logging level and have more information?

Is there anything interesting in the Event Viewer under the Application category regarding osquery? Maybe a crash report?

**Comment by christmasliu** [comment]
Thanks for checking this. As we register an own logger_plugin and in ExtensionProcessor::process_call, the extension tracks whether the request contains "status" and log the detail by itself. after add --verbose flag, could notice some logs (VLOG) like "Refreshing configuration state" in the "status" log, but the LOG(INFO) or LOG(ERROR) logs are not received at all. So there is nothing too much recorded.

**Comment by christmasliu** [comment]
Finally, got some OSQuery side logs and found the issue is caused by "Maximum sustainable CPU utilization limit exceeded", so watcher stops the worker and unfortunately, it could not restart the worker again as the error "Cannot activate testPlugin config plugin: Unknown registry plugin: testPlugin". "testPlugin" is our extended config plugin. There are 2 questions here:
1. In the user guide of OSQuery "https://osquery.readthedocs.io/en/stable/installation/cli-flags/#daemon-control-flags", the default CPU usage limit is 25% (for 9 seconds), but after checked the codes, looks current default CPU limit is 10% (for 12 seconds) defined in the file "https://github.com/osquery/osquery/blob/master/osquery/core/watcher.cpp" ({WatchdogLimitType::UTILIZATION_LIMIT, {10, 5, 100}}, and {WatchdogLimitType::LATENCY_LIMIT, {12, 6, 1000}},). Whether the user guide needs to be updated?
2. Found notice the watcher always start the worker correctly for the first time, why when it tries to restarted the worker, meet such issue "Cannot activate testPlugin config plugin: Unknown registry plugin: testPlugin". Could you please have a check for this? Thanks.

**Comment by Smjert** [comment]
1. Yes you're right, the wiki has to be updated, there's also this issue mentioning the problem: https://github.com/osquery/osquery/issues/4300. I've worked a little on this but I need to find some time to finish ^^'.

2. A config plugin is expected pretty soon in osquery start procedure, if it's coming from an extension it means that such extension should register itself when osquery starts to provide such plugin.
If that doesn't happen, osquery will fail to start.
If you start the extension manually you might want to take a look at `--extensions_require` which make osquery wait for an extension with the provided name to be registered before starting. There's an interval + timeout for this wait, you might want to have a look at the doc here: https://osquery.readthedocs.io/en/latest/installation/cli-flags/

**Comment by christmasliu** [comment]
Thanks for your comments, Smjert.
1. Besides the command line arg "watchdog_utilization_limit" to customize the CPU usage limit, could OSQuery also provides another arg to customize the duration time. For example, change the default 12s to another value.
2. The "extensions_require" was already set and the "extensions_timeout" was also set as 60 in current program. Actually in most of time, the worker process could be restarted by the watch dog process correctly but sometimes, it was failed to be restarted with the error "Cannot activate testPlugin config plugin: Unknown registry plugin: testPlugin". Noticed this once when the CPU usage limit was met and the worker process was stopped, then the issue was reproduced.

**Comment by Smjert** [comment]
1. Would it be possible to open a feature request to track this separately? Thanks!

2. Then it seems there's an issue we should look into. Just to be sure, how the extension is launched?
Would you be able to provide the osquery config and flagfile (if used)?

**Comment by christmasliu** [comment]
1. Sure. Opened another ticket to track the feature request: https://github.com/osquery/osquery/issues/7212

2. Please check the OSQuery startup arguments to launch the osqueryd "--extensions_require="xxx" --config_refresh=60 --config_plugin=testPlugin --logger_plugin=testLogger --extensions_socket=xxx --extensions_timeout=60 --database_path=xxx --pidfile osquery.pidfile --logger_path=xxx ". There are additional query configs and virtual table definitions but it contains product information and I am sorry I cannot share.
